File: packages/shared-types/src/job-status.ts
import { z } from "zod";
import { analysisResultSchema } from "./analysis-result.js";
import { explanationResponseSchema } from "./explanation.js";

export const jobStatusSchema = z.enum(["queued", "processing", "completed", "failed"]);

export const jobResultSchema = z.object({
  status: jobStatusSchema,
  result: analysisResultSchema.optional(),
  explanation: explanationResponseSchema.optional(),
  error: z.string().optional(),
});

export type JobStatus = z.infer<typeof jobStatusSchema>;
export type JobResult = z.infer<typeof jobResultSchema>;

File: apps/mobile/lib/features/analysis/domain/job_status.dart
import 'package:oralscan_ai/features/analysis/domain/analysis_result.dart';

enum JobStatus { queued, processing, completed, failed }

class JobStatusResponse {
  const JobStatusResponse({
    required this.status,
    this.result,
    this.explanation,
    this.error,
  });

  final JobStatus status;
  final AnalysisResult? result;
  final Explanation? explanation;
  final String? error;

  factory JobStatusResponse.fromJson(Map<String, dynamic> json) {
    final statusStr = json['status'] as String? ?? 'queued';
    JobStatus status;
    try {
      status = JobStatus.values.byName(statusStr);
    } catch (_) {
      status = JobStatus.queued;
    }
    AnalysisResult? result;
    if (json['result'] != null) {
      result = AnalysisResult.fromJson(json['result'] as Map<String, dynamic>);
    }
    Explanation? explanation;
    if (json['explanation'] != null) {
      explanation = Explanation.fromJson(json['explanation'] as Map<String, dynamic>);
    }
    return JobStatusResponse(
      status: status,
      result: result,
      explanation: explanation,
      error: json['error'] as String?,
    );
  }
}
